packages/payment-detection/src/thegraph/index.ts

```diff
@@ -9,24 +10,29 @@ const BASE_URL = `https://api.thegraph.com`;
 /**
  * A GraphQL client to query Request's subgraph.
  */
-export type TheGraphClient = ReturnType<typeof getSdk>;
+export type TheGraphClient<
+  TGraphClientVariant extends 'near' | null = null
+> = TGraphClientVariant extends 'near' ? ReturnType<typeof getSdkNear> : ReturnType<typeof getSdk>;
```

Comment:
@benjlevesque , I'd like your opinion about this. It may look overkill but the queries interfaces and possibilities between EVM and Near payments are very different.